inst/make_hexsticker.R

```diff
@@ -0,0 +1,253 @@
+  n = n_history_delay
+)
+
+# Make the delay PMF a data frame
+delay_df <- data.frame(
+  delay = 0:(length(delay_pmf) - 1),
+  pmf = delay_pmf
+)
+
+# Apply the delay PMF to the reporting triangle to get point nowcasts
+point_nowcast_matrix <- apply_delay(
+  rep_tri_to_nowcast = reporting_triangle,
+  delay_pmf = delay_pmf
+)
+
+# Add the point nowcast matrix to the target data
+point_nowcast_df <- target_data |>
+  mutate(nowcast = rowSums(point_nowcast_matrix))
+
+# Preprocess the training data for nowcasting
+prep_latest_data <- latest_training_data |>
+  mutate(type = "Real-time data") |>
+  select(type, reference_date, count = confirm)
+
+# Create list of truncated reporting triangles
+trunc_rep_tri_list <- truncate_triangles(
+  reporting_triangle,
+  n = n_retrospective_nowcasts
+)
+
+# Generate reporting triangles
+retro_rep_tri_list <- generate_triangles(trunc_rep_tri_list)
+
+# Generate retrospective point nowcasts
+retro_pt_nowcast_mat_list <- generate_pt_nowcast_mat_list(
+  reporting_triangle_list = retro_rep_tri_list,
+  n = n_history_delay
+)
+
+# Estimate a vector of negative binomial dispersion parameters from the observations and estimates at each horizon
+disp_params <- estimate_dispersion(
+  pt_nowcast_mat_list = retro_pt_nowcast_mat_list,
+  trunc_rep_tri_list = trunc_rep_tri_list,
+  reporting_triangle_list = retro_rep_tri_list,
+  n = n_retrospective_nowcasts
+)
+
+# Generate multiple draws of a nowcast combining observed and predicted values
+nowcast_draws_df <- get_nowcast_draws(
+  point_nowcast_matrix,
+  reporting_triangle,
+  dispersion = disp_params,
+  draws = 100
+)
+
+# Prepare the data
+latest_data_prepped <- latest_training_data |>
+  mutate(time = row_number()) |>
+  rename(obs_confirm = confirm) |>
+  mutate(reference_date = as.Date(reference_date))
+
+# Prepare the final data with the latest confirmed counts
+final_data_prepped <- target_data |>
+  select(reference_date, final_confirm = confirm) |>
+  mutate(reference_date = as.Date(reference_date))
+
+# Join the nowcast draws with the latest observed data and final confirmed counts
+obs_with_nowcast_draws_df <- nowcast_draws_df |>
+  left_join(latest_data_prepped, by = "time") |>
+  left_join(final_data_prepped, by = "reference_date")
+
+# Combine the nowcast draws with the observed and final confirmed counts
+combined_data <- obs_with_nowcast_draws_df |>
+  select(reference_date, obs_confirm, final_confirm) |>
+  distinct() |>
+  pivot_longer(
+    cols = c(obs_confirm, final_confirm),
+    names_to = "type",
+    values_to = "count"
+  ) |>
+  mutate(type = case_when(
+    type == "obs_confirm" ~ "Observed data",
+    type == "final_confirm" ~ "Final observed data"
+  ))
+
+# Plot with draws for nowcast only
+hex_plot <- ggplot() +
+  # Add nowcast draws as thin gray lines
+  geom_line(
+    data = obs_with_nowcast_draws_df,
+    aes(
+      x = reference_date,
+      y = pred_count,
+      group = draw,
+      color = "Nowcast draw"
+    )
```

Comment:
```suggestion
    ),
    linewidth = 0.2
```
Play with some smaller lines